internal/controllers/device_data_controller.go

```diff
@@ -271,7 +271,7 @@ func (udc *UserDevicesController) RefreshUserDeviceStatus(c *fiber.Ctx) error {
 	return fiber.NewError(fiber.StatusBadRequest, "no active Smartcar integration found for this device")
 }
 
-var errorCodeRegex = regexp.MustCompile(`^[A-Z0-9]{5,8}$`)
+var errorCodeRegex = regexp.MustCompile(`^.{5,8}$`)
```

Comment:
This seems like the second major change here, removing validation outside of length.